### PR TITLE
Fix stub-only partial namespace packages not recognized as packages

### PR DESCRIPTION
**Pull request by henribru**
Resolves: https://github.com/python-poetry/poetry/issues/4621

- [x] Added **tests** for changed code.
- N/A Updated **documentation** for changed code.

The tests are based on the existing tests for stub-only packages introduced in https://github.com/python-poetry/poetry/pull/2000

### CODE REVIEW

**Review by neersighted — DISMISSED** [review]

**Comment by abn** [comment]
@henribru can you rebase this please?

**Comment by henribru** [comment]
Done!

File: tests/masonry/utils/test_package_include.py [hunk 1]
```diff
@@ -65,6 +65,20 @@ def test_pep_561_stub_only_package_good_name_suffix() -> None:
     ]
 
 
+def test_pep_561_stub_only_partial_namespace_package_good_name_suffix() -> None:
+    pkg_include = PackageInclude(
+        base=fixtures_dir / "pep_561_stub_only_partial_namespace", include="good-stubs"
+    )
+    print(pkg_include.elements)
```
Comment (abn) [inline tests/masonry/utils/test_package_include.py:72]:
```suggestion

```

Comment (henribru) [reply tests/masonry/utils/test_package_include.py]: Thanks, didn't notice I had left that lying around

**Review by neersighted — APPROVED** [review]

**Pull request merged**